### PR TITLE
Documentation for Debugging Snapshotter

### LINKED ISSUE #6309: Cluster Autoscaler - Add docs for debugging snapshotter
**Which component are you using?**:

Cluster-autoscaler

**Is your feature request designed to solve a problem? If so describe the problem this feature should solve.**:

We implemented the [debugging snapshotter](https://github.com/kubernetes/autoscaler/pull/4552) feature for the cluster autoscaler to provide significant help to users when trying to understand the CA's behaviour. 

We've previously demoed this at [Kubecon EU 2023](https://youtu.be/VgTWf4mjyG8?si=iHkcv5FT2yO3q5x0&t=596), however currently we don't have any docs for this feature in the repo to help users discover and use it.

**Describe the solution you'd like.**:

Docs added to explain the use of the debugging snapshotter.

**Describe any alternative solutions you've considered.**:

N/A

**Additional context.**:

### PR DESCRIPTION
**Pull request by Dharma-09**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
Documents that explain how to use the debugging snapshotter.  

#### Which issue(s) this PR fixes:

Fixes #6309 

#### Does this PR introduce a user-facing change?
NONE

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cluster-autoscaler/debuggingsnapshot/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	Error                         string                  `json:"Error,omitempty"`
+	StartTimestamp                time.Time               `json:"StartTimestamp"`
+	EndTimestamp                  time.Time               `json:"EndTimestamp"`
+	TemplateNodes                 map[string]*ClusterNode `json:"TemplateNodes"`
+}
+
+```
+
+#### Development
+1. First step is enable the snapshotter adding the following flag to cluster-autoscaler [manifest](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/bizflycloud/manifest/cluster-autoscaler.yaml) .
```
Comment (gjtempleton) [inline cluster-autoscaler/debuggingsnapshot/README.md:24]: We should avoid referring to a single cloud provider implementation's of the manifest.
```suggestion
1. First step is to enable the snapshotter feature by adding the following flag to cluster-autoscaler.
```

File: cluster-autoscaler/debuggingsnapshot/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+```
+2. Save the updated manifest file and apply it to your Kubernetes cluster using:
+
+```bash
+kubectl apply -f <path-to-your-manifest-file>
+```
+3. once you deploy your pod make a snap shot request
+```sh
+ curl http://127.0.0.1:8085/snapshotz > FIlE_NAME.json
+```
```
Comment (gjtempleton) [inline cluster-autoscaler/debuggingsnapshot/README.md:37]: This is all very tied to users deploying via raw manifests, we should avoid going into this level of detail in these documents and assume readers know how to add a command line flag to their method of deployment, whatever that is.

**Review by gjtempleton — COMMENTED** [review]
Thanks for raising this, I've got a couple of quick concerns, I think we should focus this more on how to navigate the data the snapshotter provides to users rather than getting lost in the details of how to set the flag to true.

File: cluster-autoscaler/debuggingsnapshot/README.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
+# Debugging Snapshotter
+ It's a tool to visualize the internal state of cluster-autoscaler at a point in time to help debug autoscaling issues.
+
+---
+### Requirements
+Require Cluster-autoscaler versions 1.24+
+
+---
+#### What data snapshotter can capture?
+https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/debuggingsnapshot/debugging_snapshot.go#L60C1-L71C1
```
[the remark this answers is not in the document]
Comment (Dharma-09) [reply cluster-autoscaler/debuggingsnapshot/README.md:10]: Okay, I'll change it to permalink

File: cluster-autoscaler/debuggingsnapshot/README.md (hunk 2 above)
Comment (Dharma-09) [reply cluster-autoscaler/debuggingsnapshot/README.md:37]: I have one question. Is there anything additional I need to do apart from adding the flag instruction?

Thank you

File: cluster-autoscaler/debuggingsnapshot/README.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	EndTimestamp                  time.Time               `json:"EndTimestamp"`
+	TemplateNodes                 map[string]*ClusterNode `json:"TemplateNodes"`
+}
+
+```
+## Development
+Add the following flag to your cluster-autoscaler configuration to enable the snapshotter feature.
+```
+--debugging-snapshot-enabled=true
+```
```
Comment (gjtempleton) [inline cluster-autoscaler/debuggingsnapshot/README.md:26]: Can we also give a one line instruction on how access the snapshot from the command line as well?

**Comment by gjtempleton** [comment]
Thanks, great to finally have docs on this to point users to!

/label tide/merge-method-squash
/lgtm

**Pull request merged**